src/app/app/app-effects/effects/map/visualizers.app.effects.ts

```diff
@@ -191,6 +193,62 @@ export class VisualizersAppEffects {
 			});
 		});
 
+	/**
+	 * @type Effect
+	 * @name drawDistamceMeasureOnMap$
+	 * @ofType DrawOverlaysOnMapTriggerAction
+	 * @dependencies overlays, cases
+	 */
+	@Effect({ dispatch: false })
+	drawDistamceMeasureOnMap$: Observable<any> = this.actions$
+		.ofType<SetMeasureDistanceToolState>(ToolsActionsTypes.SET_MEASURE_TOOL_STATE)
+		.withLatestFrom(this.store$.select(mapStateSelector), (action, mapState: IMapState) => [action, mapState])
+		.map(([action, mapState]: [SetMeasureDistanceToolState, IMapState]) => {
+			return [action, MapFacadeService.activeMap(mapState)];
+		})
+		.filter(([action, activeMap]: [SetMeasureDistanceToolState, CaseMapState]) => Boolean(activeMap))
+		.map(([action, activeMap]: [SetMeasureDistanceToolState, CaseMapState]) => {
+			const communicator = this.imageryCommunicatorService.provide(activeMap.id);
+			const distanceVisualizerTool = <MeasureDistanceVisualizer>communicator.getVisualizer(MeasureDistanceVisualizerType);
+			if (distanceVisualizerTool) {
+				if (action.payload) {
+					distanceVisualizerTool.createInteraction();
+				} else {
+					distanceVisualizerTool.removeInteraction();
```

Comment:
removeInteraction + clearEntities always come together (for now)
can be called as a single function e.g. distanceVisualizerTool.clear()